### PR TITLE
Update flow to 0.180.0

### PR DESCRIPTION
**Pull request by mischnic**
...

### CODE REVIEW

File: packages/utils/node-resolver-core/src/NodeResolver.js [hunk 1]
```diff
@@ -1149,8 +1146,6 @@ export default class NodeResolver {
           sourcePath: pkg.pkgfile,
           resolved: alias.global,
         };
-      } else if (alias.fileName) {
-        alias = alias.fileName;
```
Comment (mischnic) [inline packages/utils/node-resolver-core/src/NodeResolver.js:1153]: Flow said that this is unreachable type-wise. And this branch was indeed never reached in the tests. So either it really is dead, or we don't test this at all.

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.96s | +13.00ms |
| Cached | 423.00ms | -16.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 97.00ms | +6.00ms ⚠️ |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 98.00ms | +6.00ms ⚠️ |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 99.00ms | +7.00ms ⚠️ |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 12.08s | -74.00ms |
| Cached | 554.00ms | -3.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.14m | -1.08s |
| Cached | 3.15s | +66.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 8.71s | -71.00ms |
| Cached | 357.00ms | +7.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/334369382697271890)**

File: packages/core/core/src/Environment.js [hunk 2]
```diff
@@ -20,18 +20,20 @@ type EnvironmentOpts = {|
...
+    engines,
+    includeNodeModules,
+    outputFormat,
+    sourceType = 'module',
+    shouldOptimize = false,
+    isLibrary = false,
+    shouldScopeHoist = false,
+    sourceMap,
+    loc,
+  }: EnvironmentOpts = {...null},
```
Comment (devongovett) [inline packages/core/core/src/Environment.js]: This is weird. What's going on here? All the properties are optional right, so why null?

Comment (mischnic) [reply packages/core/core/src/Environment.js]: This is not about the properties, it's literally only about exact/inexact/unsealed objects: https://github.com/facebook/flow/issues/7424

https://flow.org/try/#0DYUwLgBAHgXBDeAfRBfCBeBKDcAoUkAnnPGpqXgRAF4nJkIB0zAdgK7DA5A

https://github.com/facebook/flow/issues/2977

Comment (devongovett) [reply packages/core/core/src/Environment.js]: That's super weird when all the properties are optional. Therefore an empty object should be valid. I guess `{...null}` has some additional runtime overhead. _sigh_

Comment (mischnic) [reply packages/core/core/src/Environment.js]: It gets weirder: https://flow.org/try/#0DYUwLgBAHgXBDeAfAhgfjgOwK4FsBGIATogL4QC8CJA3BAPR0RgAWIEAxloYSBpAA6EA9nlA4AUJNCQAnnHjJMuAoTKV4NeoxZscyAOYBLdhCEAzCFgwBnEMlAATU3gBWIdmGtA

Comment (mischnic) [reply packages/core/core/src/Environment.js]: Alright, `{/*:: ...null */}` works as well, and doesn't do anything at runtime. I've also replaced all preexisting occurrences of `{...null}` with the comment version.

**Review by devongovett — APPROVED** [review]

File: packages/core/core/src/Environment.js (hunk 2 above)
Comment (devongovett) [reply packages/core/core/src/Environment.js]: lol wow

**Pull request merged**